app/rest/rest/src/main/java/io/syndesis/rest/metrics/PrometheusMetricsProviderImpl.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.rest.metrics;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.internal.readiness.Readiness;
+import io.syndesis.model.metrics.IntegrationMetricsSummary;
+import io.syndesis.rest.metrics.prometheus.HttpClient;
+import io.syndesis.rest.metrics.prometheus.HttpQuery;
+import io.syndesis.rest.metrics.prometheus.QueryResult;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.stereotype.Component;
+
+@Component
+@ConditionalOnProperty(value = "metrics.kind", havingValue = "prometheus")
+@EnableConfigurationProperties({PrometheusConfigurationProperties.class})
+public class PrometheusMetricsProviderImpl implements MetricsProvider {
+
+    private final String serviceName;
+    private final KubernetesClient kubernetes;
+
+    private final HttpClient httpClient = new HttpClient();
+
+    protected PrometheusMetricsProviderImpl(PrometheusConfigurationProperties config, KubernetesClient kubernetes) {
+        this.serviceName = config.getService();
+        this.kubernetes = kubernetes;
+    }
+
+    @Override
+    public IntegrationMetricsSummary getIntegrationMetricsSummary(String integrationId) {
+
+        // map integration Id to pod name
+        List<Pod> integrationPodList = kubernetes.pods().withLabel("integration").list().getItems();
+        final List<Pod> pods = integrationPodList.stream()
+            .filter(p -> Readiness.isReady(p)
+                && integrationId.equals(p.getMetadata().getAnnotations().get("syndesis.io/integration-id")))
+            .collect(Collectors.toList());
+
+        if (pods.isEmpty()) {
```

Comment:
Minor suggestion: Instead of getting a list of pods and checking if its empty, you could just use `findFirst().orElseThrow(()-> new IllegalArgumentException("Missing pod with integration id" + integrationId)` directly on the stream.

Reply:
Hi @iocanel that makes sense. I left it as a list since we would also have to handle the case when user wants to run replicas of the integration in the future. 
That might not be in scope today, but has to be handled later. Should have added a comment.